detox/src/utils/environment.js

```diff
@@ -187,7 +187,31 @@ async function getFrameworkPath() {
 async function _doGetFrameworkPath() {
   const detoxVersion = getDetoxVersion();
   const sha1 = (await exec(`(echo "${detoxVersion}" && xcodebuild -version) | shasum | awk '{print $1}'`)).stdout.trim();
-  return `${DETOX_LIBRARY_ROOT_PATH}/ios/${sha1}/Detox.framework`;
+  return `${DETOX_LIBRARY_ROOT_PATH}/ios/framework/${sha1}/Detox.framework`;
+}
+
+let _iosXCUITestRunnerPath;
+
+async function getXCUITestRunnerPath() {
+  if (!_iosXCUITestRunnerPath) {
+    _iosXCUITestRunnerPath = _doGetXCUITestRunnerPath();
+  }
+
+  return _iosXCUITestRunnerPath;
+}
+
+async function _doGetXCUITestRunnerPath() {
+  const detoxVersion = getDetoxVersion();
+  const sha1 = (await exec(`(echo "${detoxVersion}" && xcodebuild -version) | shasum | awk '{print $1}'`)).stdout.trim();
+
+  const derivedDataPath = `${DETOX_LIBRARY_ROOT_PATH}/ios/xcuitest-runner/${sha1}`;
+
+  const xctestrunPath = (await exec(`find ${derivedDataPath} -name "*.xctestrun" -print -quit`)).stdout.trim();
```

Comment:
Is it because the path is dynamic (random, guidish something)?

Reply:
yeah, it can be dynamic based on the env.. e.g.
`DetoxXCUITestRunner_DetoxXCUITestRunner_iphonesimulator17.4-arm64-x86_64.xctestrun`